metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/ScriptParser.java

```diff
@@ -0,0 +1,189 @@
+	protected String timestampField;
+	protected SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.S z");
+
+	@Override
+	public void configure(Map<String, Object> config) {
+		this.scriptPath=(String) config.get("path");
+		this.parseFunction=(String)config.get("function");
+		this.language=(String)config.get("language");
+		this.commonScript=this.commonScript+language+"/common";
+		if(this.parseFunction==null)
+			this.parseFunction="parse";
+	}
+	//Should this be sent to the interface as a default method?
+	public InputStream openInputStream(String streamName) throws IOException {
+	    FileSystem fs = FileSystem.get(new Configuration());
+	    Path path = new Path(streamName);
+	    if(fs.exists(path)) {
+	      return fs.open(path);
+	    } else {
+	      return getClass().getResourceAsStream(streamName);
+	    }
+	  }
+
+	@Override
+	public void init() {
+		engine = new ScriptEngineManager().getEngineByName(this.language);
+		try{
+			InputStream commonStream = openInputStream(this.commonScript);
+			if (commonStream == null) {
+		        throw new RuntimeException(
+		                "Unable to initialize "+this.language+" Parser: Unable to load " + this.commonScript + " from either classpath or HDFS");
+		      }
+
+		      engine.eval(new InputStreamReader(commonStream));
+		      if (LOG.isDebugEnabled()) {
+		        LOG.debug("Loading parser-specific functions from: " + this.scriptPath);
+		      }
+
+		      InputStream patterInputStream = openInputStream(this.scriptPath);
+		      if (patterInputStream == null) {
+		        throw new RuntimeException("Script parser unable to initialize "+this.language+" parser: Unable to load " + this.scriptPath
+		                + " from either classpath or HDFS");
+		      }
+		      engine.eval(new InputStreamReader(patterInputStream));
+		}catch(Throwable e){
+			LOG.error(e.getMessage(), e);
+		    throw new RuntimeException(this.language+" Script parser Error: " + e.getMessage(), e);
+		}
+		
+		Invocable invocable = (Invocable) engine;
+		
+	}
+
+	@Override
+	public List<JSONObject> parse(byte[] rawMessage) {
+		List<JSONObject> messages = new ArrayList<JSONObject>();
+		if(engine==null)
+			init();
+		String originalMessage = null;
+		try{
+			originalMessage = new String(rawMessage, "UTF-8");
+			if (LOG.isDebugEnabled()) {
+	          LOG.debug("Script parser parsing message: " + originalMessage);
+	        }
+			Invocable invocable = (Invocable) engine;
+	
+			Object result = invocable.invokeFunction(this.parseFunction,originalMessage);
+			JSONObject message = new JSONObject((Map)result);
+			
+			if (message.size() == 0)
+		        throw new RuntimeException("Script produced a null message. Original message was: "
+		                + originalMessage + " and the parsed message was: " + message + " . Check the function at: "
+		                + this.scriptPath);
+
+		      message.put("original_string", originalMessage);
```

Comment:
To move toward consistency, can this "original_string" field be updated to use the new `Constants.Fields.ORIGINAL.getName()`?